### PR TITLE
Test 2

### PR DESCRIPTION
**Pull request by lerfast**
Hi, we did the following task for the part 2 of the exercise:

- [ ] We write tests for the remaining functions related to status and content updates, as well as the re-positioning of items in the list. 
- [ ] We use the Jest testing framework and create separate describe blocks for each function.
- [ ] We write test cases to check if the functions perform the desired behavior.

### CODE REVIEW

File: src/utils.test.js [hunk 1]
```diff
@@ -8,172 +8,181 @@ import {
...
+  localStorage.setItem('todoList', JSON.stringify(initialList));
+  const updatedList = deleteTask(2, initialList);
+  const expectedList = [
+    { id: 1, description: 'Task 1', completed: false },
+    { id: 3, description: 'Task 3', completed: false },
+  ];
+  expect(updatedList).toHaveLength(2);
+  expect(updatedList).toEqual(expectedList);
+});
+test('clearCompletedTasks should clear completed tasks from the list', () => {
```
Comment (shahadat3669) [inline src/utils.test.js]: 👍 Your test cases have surpassed my expectations, showing a high level of attention to detail. However, there's an area that needs improvement:

- To enhance the organization and readability of your test suite, you should group your tests using the [describe()](https://jestjs.io/docs/api#describename-fn) method. By creating logical blocks using the `describe` blocks, you can group related test cases together, making it easier to understand the test suite's structure. Grouping tests into descriptive blocks allows for better management and maintenance, especially when you need to focus on specific methods or classes. Following this approach aligns with the project requirements and will improve the overall clarity of your test suite.

image


Keep up the great work with your test cases, and I'm confident that organizing them using `describe` blocks will take your test suite to the next level! 🌟🧪🚀

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @lerfast, @CobbyElsonfx, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Descriptive PR Title and Description
✅ The code is top-notch! 💯


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/utils.test.js (hunk 1 above)
Comment (lerfast) [reply src/utils.test.js]: Did it, thanks!!!
image

**Review by Owusu-Desmond — APPROVED** [review]
Hi @lerfast and @As1imwe-Mark  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 All your tests pass✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**